libnavigation-core/src/main/java/com/mapbox/navigation/core/DeveloperMetadataObserver.kt

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+
+/**
+ * Callback that provides [DeveloperMetadata].
+ * Can be registered via [MapboxNavigation.registerDeveloperMetadataObserver]
+ * and unregistered via [MapboxNavigation.unregisterDeveloperMetadataObserver].
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+interface DeveloperMetadataObserver {
```

Comment:
```suggestion
fun interface DeveloperMetadataObserver {
```

Reply:
I see that there are a lot of fun interfaces among the API but why exactly do you make them `fun`? To use lambdas when implementing them?

Reply:
Correct [Functional (SAM) interfaces](https://kotlinlang.org/docs/fun-interfaces.html) FTW